### DISCUSSION TITLE
Is there a mechanism for multi 'toolkit' libraries?

### DISCUSSION THREAD

**Discussion by nyholku** [bytedeco/javacpp#544] [Q&A] [answered]
I tried to look at documentation and some example code but did not spot this.

The library I'm working on, Open Cascade or OCC, is made of number of 'toolkits' each with their own shared library. 

I could just lump everything together but the load time for those native libs seems to be quite a few seconds so would be better if only those toolkits that are actually used get loaded. OCC is one of the biggest and best organised C++ libraries I've seen containing some ~3000 classes and files.

More importantly while developing OCC bindings with JavaCPP it would be nice and a great time saver it everything was not just lumped together.

So is there a mechanism do something like this:

Compile/generate a binding for the base library and then in a separate compile/generate step refer to that so that the references to classes in the base library would be picked up from there.

I hacked JavaCPP a bit and it seems that doing this would not be too difficult, before it dawned on me that I should have asked first if there is something like that already in place.

If there is not then I'm willing to have a look at developing such a mechanism.

**Comment by saudet** [comment, marked as answer]
We can of course do that, yes. You can check what it looks like for OpenCV, FFmpeg, CUDA, etc:
https://github.com/bytedeco/javacpp-presets/tree/master/opencv/src/main/java/org/bytedeco/opencv/presets
https://github.com/bytedeco/javacpp-presets/tree/master/ffmpeg/src/main/java/org/bytedeco/ffmpeg/presets
https://github.com/bytedeco/javacpp-presets/tree/master/cuda/src/main/java/org/bytedeco/cuda/presets
